File: scotch-score/src/types/scotch.ts
import CurrencyCode from './currencyCode';
import ScotchRegion from './scotchRegion';

interface Scotch {
  id: string;
  name: string;
  description: string;
  distillery: string;
  region: ScotchRegion;
  age: number;
  amount: number;
  currency: CurrencyCode;
  images: string[];
  ratingCounts: number[];
  averageRating: number;
  dateCreated: Date;
}

export default Scotch;

File: ScotchScore.Contracts/Scotch.cs
using System;
using System.Collections.Generic;

namespace ScotchScore.Contracts
{
    public class Scotch
    {
        public int Age { get; set; }
        public decimal Amount { get; set; }
        public decimal AverageRating { get; set; }
        public CurrencyCode Currency { get; set; }
        public DateTime DateCreated { get; set; }
        public string Description { get; set; }
        public string Distillery { get; set; }
        public string Id { get; set; }
        public IReadOnlyList<string> Images { get; set; }
        public string Name { get; set; }
        public int[] RatingCounts { get; set; }
        public Region Region { get; set; }
    }
}